engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -210,6 +210,30 @@ Object doFunctionalDispatchUncachedSymbol(
     Type selfTpe = typesLibrary.getType(self);
     Function function = resolveFunction(symbol, selfTpe, methodResolverNode);
     if (function == null) {
+      if (self instanceof Type t && isNamedAsAssociatedType(t)) {
+        CompilerDirectives.transferToInterpreter();
+        // if we represent a Module
+        var scope = t.getDefinitionScope();
+        var optionType = scope.getType(symbol.getName());
+        if (optionType.isPresent()) {
+          return optionType.get();
+        }
+        var module = scope.getModule();
+
+        var ctx = EnsoContext.get(this);
+        var moduleName =
+            module.getName().item().equals("Main")
+                ? module.getPackage().libraryName().toString()
+                : module.getName().toString();
+        var subModuleName = moduleName + "." + symbol.getName();
```

Comment:
This is correct. We are in *FQN* resolution, so we are looking only for submodules (child modules). We are not looking for symbols that are exported from the module.

Reply:
Hmm, I don't see why we can be sure here that we are not deeper, but I'm just not familiar enough with this code. I may try to understand it deeper then.

Anyway - if we are so sure, let's add an assert here `assert symbol.getName().path().length() == 2` - that would make it clearer that we know what we are doing (which you probably do know, but I as a reader trying to get familiar with this part of the codebase - don't; and without such pointers it's much harder to get acquainted with it).

Reply:
Done in 177d402c